public/src/overmind/actions.tsx

```diff
@@ -165,32 +166,93 @@ export const setEnrollmentState = async ({ effects }: Context, enrollment: Enrol
 /** Updates a given submission with a new status. This updates the given submission, as well as all other occurrences of the given submission in state. */
 export const updateSubmission = async ({ state, effects }: Context, { owner, submission, status }: { owner: SubmissionOwner, submission: Submission | null, status: Submission_Status }): Promise<void> => {
     /* Do not update if the status is already the same or if there is no selected submission */
-    if (!submission || submission.status === status) {
+    if (!submission) {
         return
     }
 
+    switch (owner.type) {
+        // Take no action if there is no change in status
+        case "ENROLLMENT":
```

Comment:
Should we use a `const` for the strings `ENROLLMENT` and `GROUP` to ensure that we don't mistakenly introduce a typo, when it is used multiple times.

Reply:
In this case the `type` field is defined using string literal types "ENROLLMENT" | "GROUP". Any other string will give an error.